File: Faza6/OnlineGamesShop/src/main/java/com/rich/onlinegamesshop/model/views/UserLibraryView.java
package com.rich.onlinegamesshop.model.views;

import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.Immutable;

import java.math.BigDecimal;
import java.time.LocalDate;

@Data
@Entity
@Immutable
@Table(name = "userlibraryview")
@NoArgsConstructor
@AllArgsConstructor
public class UserLibraryView {
    @Id
    private String title;
    private String genre;
    private BigDecimal price;
    private String developer;
    private int stock;
    private LocalDate releaseDate;
    private BigDecimal discountPercentage;
    private LocalDate startDate;
    private LocalDate endDate;
    private BigDecimal amount;
    private String paymentMethod;
    private LocalDate date;
}

File: Faza6/FrontEnd/Angular/src/app/model/UserLibraryView.ts
import Big from "big.js";
export class UserLibraryView{
  title:string;
  genre:string;
  price:Big;
  developer:string;
  stock:number;
  releaseDate:string;
  discountPercentage:Big;
  startDate:string;
  endDate:string;
  amount:Big;
  paymentMethod:string;
  date:string;
  constructor(
    title:string,
    genre:string,
    price:Big,
    developer:string,
    stock:number,
    releaseDate:string,
    discountPercentage:Big,
    startDate:string,
    endDate:string,
    amount:Big,
    paymentMethod:string,
    date:string,
  ){
      this.title=title;
      this.genre=genre;
      this.price=price;
      this.developer=developer;
      this.stock=stock;
      this.releaseDate=releaseDate;
      this.discountPercentage=discountPercentage;
      this.startDate=startDate;
      this.endDate=endDate;
      this.amount=amount;
      this.paymentMethod=paymentMethod;
      this.date=date;
  }
}
